contrib/epee/include/net/http_protocol_handler.inl

```diff
@@ -208,11 +208,45 @@ namespace net_utils
 		m_newlines(0),
 		m_bytes_read(0),
 		m_psnd_hndlr(psnd_hndlr),
-		m_conn_context(conn_context)
+		m_conn_context(conn_context),
+		m_initialized(false)
 	{
 
 	}
 	//--------------------------------------------------------------------------------------------
+	template<class t_connection_context>
+	simple_http_connection_handler<t_connection_context>::~simple_http_connection_handler()
+	{
+	  try
+	  {
+	    if (m_initialized)
+	    {
+	      CRITICAL_REGION_LOCAL(m_config.m_lock);
+	      m_config.m_connection_count -= std::min(std::size_t(1), m_config.m_connection_count);
```

Comment:
I think you could remove the `min` by pushing this into the `if(connection found)`.

Reply:
Converted this into a simple `if (m_connection_count) --m_connection_count`